okhttp/src/jvmMain/kotlin/okhttp3/ResponseBody.kt

```diff
@@ -182,14 +100,15 @@ abstract class ResponseBody : Closeable {
    * may trigger an [OutOfMemoryError]. Prefer to stream the response body if this is a
    * possibility for your response.
    */
+  // TODO consider which of these should be part of common API
```

Comment:
I think we include this in the common API, but throw an exception if the charset is specified but is not UTF-8.

Reply:
It's a bit nasty that UnsupportedEncodingException is an IOException, likely to cause retries rather than early failure.

Reply:
On JVM we currently fallback to default, which seems safe there, but not on non JVM.

```
  fun charset(defaultValue: Charset? = null): Charset? {
    val charset = parameter("charset") ?: return defaultValue
    return try {
      Charset.forName(charset)
    } catch (_: IllegalArgumentException) {
      defaultValue // This charset is invalid or unsupported. Give up.
    }
  }
```

Reply:
Hmmm... falling back to UTF-8 universally feels bad to me. I don't really like our JVM behavior either.

I see what you're saying regarding IOExceptions. But for non-JVM... I still think our least bad option is to throw an IOException. I don't think a RuntimeException is appropriate either; the client shouldn't crash if it can't understand the server.